pkg/assets/store.go

```diff
@@ -217,6 +217,22 @@ func (s *Store) AddBearerToken(ctx context.Context, ns string, sel v1.SecretKeyS
 	return nil
 }
 
+func (s *Store) AddAuthorizationCredentials(ctx context.Context, namespace string, auth *monitoringv1.Authorization, key string) error {
+	if auth == nil {
+		return nil
+	}
+
+	if err := auth.Validate(); err != nil {
+		return err
+	}
+
+	err := s.AddBearerToken(ctx, namespace, *auth.Credentials, key)
```

Comment:
Since we are now using this function to add different token types to it, should we rename it to `AddToken` and the underlying map to `TokenAssets`?